doc/DataPlaneCodeGeneration/DeveloperDrivenEvolution.md

```diff
@@ -97,7 +97,48 @@ public virtual async Task<Response<MetricFeedback>> GetMetricFeedbackValuesAsync
 If the suggested rename does not apply to your scenario, you could pick the best name fitting your API.
 
 ## Implement the method
-You could call the protocol method inside your convenience method and map the returned raw response to model. See different scenarios and corresponding examples below. 
+To implement method body, go through the following steps:
+
+1. [Optionally open a `DiagnosticScope`](#optionally-open-a-diagnosticscope).
+2. [Serialize any input model to `RequestContent`](#serialize-any-input-model-to-requestcontent)
+3. [Call the protocol method](#call-the-protocol-method)
+4. [Deserialize the `Response` to response with output model](#deserialize-the-response-to-response-with-output-model)
+
+### Optionally open a `DiagnosticScope`
+In the generated client constructor, the `ClientDiagnostics` is already initialized as 
+```C#
+ClientDiagnostics = new ClientDiagnostics(options, true);
+```
+which means the `DiagnosticScope` who is not the outmost one will be suppressed, and only the outmost one will be left in the telemetry.
+
+The generated protocol method opens a diagnostic scope for each operation. This scope can be used by the convenience method in many cases. Open a new `DiagnosticScope` only when:
+- The convenience method has a different name than the protocol method.
+- There is significant logic in the convenience method beyond model serialization/deserialization.
+- If you need to add attributes or otherwise modify the diagnostic scope.
+
+If a new diagnostic scope is added in the convenience method, the one in the protocol method should be suppressed so that there is only one distributed tracing span corresponding to the public method invocation. This should happen by default in generated clients, where `suppressNestedClientActivities` is set to `true` in the `ClientDiagnostics` constructor.
```

Comment:
Will you put the sample code here to be consistent with other section?

using var scope = ClientDiagnostics.CreateScope("MetricsAdvisorClient.GetMetricFeedbackValue");

Reply:
This code should be in the method, but here we are talking about the constructor. The code in the constructor should be like Line 110: ClientDiagnostics = new ClientDiagnostics(options, true);